### PR TITLE
fluxcd: fix wrong manifest hash; add superherointj as maintainer

### LINKED ISSUE #135539: fluxcd 0.16.2 broken
### Describe the bug
Bootstrap does not create the correct versions of imagerepositories. See https://github.com/fluxcd/flux2/issues/759 for more detail.

### Steps To Reproduce
1. run flux bootstrap with `--components-extra=image-reflector-controller,image-automation-controller`
2. run `kubectl get crd imagerepositories.image.toolkit.fluxcd.io -oyaml`

### Expected behavior
`storedVersion` should be `v1beta1` but is `v1alpha2`

### Additional context
https://github.com/fluxcd/flux2/issues/759

### Notify maintainers
@jlesquembre 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.6.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.4pre20210503_6d2553a`
 - channels(davidsmith): `"nixos-20.09.4393.068984c00e0, nixpkgs-20.09.4393.068984c00e0, unstable-21.11pre310022.14b0f20fa1f"`
 - nixpkgs: `/Users/davidsmith/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by superherointj**
fluxcd:
* fix wrong manifest hash;
* add superherointj as maintainer

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

fixes #135539

### CODE REVIEW

File: pkgs/applications/networking/cluster/fluxcd/default.nix [hunk 1]
```diff
@@ -55,7 +55,10 @@ buildGoModule rec {
     '';
     homepage = "https://fluxcd.io";
     license = licenses.asl20;
-    maintainers = with maintainers; [ jlesquembre ];
+    maintainers = with maintainers; [
+      jlesquembre
+      superherointj
+    ];
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/fluxcd/default.nix:61]:
```suggestion
    maintainers = with maintainers; [ jlesquembre superherointj ];
```

File: pkgs/applications/networking/cluster/fluxcd/default.nix [hunk 2]
```diff
@@ -55,7 +55,10 @@ buildGoModule rec {
...
     platforms = platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/fluxcd/default.nix:62]:
```suggestion
```
buildGoModule defaults to a subset of unix.

Comment (superherointj) [reply pkgs/applications/networking/cluster/fluxcd/default.nix]: Done

File: pkgs/applications/networking/cluster/fluxcd/default.nix (hunk 1 above)
Comment (superherointj) [reply pkgs/applications/networking/cluster/fluxcd/default.nix]: Done

**Review by jlesquembre — APPROVED** [review]

**Comment by shmish111** [comment]
will test later today

**Pull request merged**